src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -187,24 +203,23 @@ private Meal pickRandomMeal(List<Meal> mealList) {
         return (Meal)mealList.get(index);
     }
 
-    private Boolean isResultOfSearch(Meal meal, List<String> params) {
+    private int getFrequency(Meal meal, List<String> params) {
```

Comment:
Now that the method has become a bit more complicated, I'd suggest adding a comment on top.

It wasn't necessary before, because the function name and return type were self explanatory, but that's no longer the case.
You can shortly describe what "frequency" is, in this context. Oftentimes a simple example serves as the best comment, e.g. 
```
getFrequency(
  new Meal(1, "Chocolate Cake", "A cake made of chocolate", ...),
  new List("chocolate", "cake")
) == <whatever you expect the result to be>
```
Pardon my Java, Bartek may know what the usual syntax for those "doctests" in Java is.